package.json

```diff
@@ -17,27 +17,27 @@
     "url": "https://github.com/typescript-eslint/typescript-eslint/issues"
   },
   "scripts": {
-    "build": "lerna run build --ignore ast-spec",
```

Comment:
This was a non-obvious one - I had attempted to emulate the intent behind this command as "build all packages apart from ast-spec" and that's indeed what my nx command was first doing...

However, this wasn't actually doing anything in lerna, because the package is called `@typescript-eslint/ast-spec` as far as lerna is concerned. Any idea how this happened @bradzacher, or why it was originally set up to try and avoid building ast-spec?

Reply:
Proof from the latest master build:

![image](https://user-images.githubusercontent.com/900523/120208534-a2cc7000-c23e-11eb-8e35-62255ccb80ec.png)

Reply:
Yeah I realised this after I committed it to master.

My follow up PR #3258 has a fix for it.

The ast-spec is automatically built as part of the types package, because it gets copied over. So the idea was that we could avoid doing it twice.

Usually wouldn't worry, but the build runs the api-extractor to create a single type file for the ast types, and it's not cached like the TS build, so it just wastes a few seconds.

Reply:
Which projects implicitly depend on ast-spec then? We need to inform Nx of those relationships (any relationship which is not via a static export + import is classed as an "implicit dependency" and we can annotate the dep graph with it in nx.json)

Reply:
Actually this has brought up a good point... 

I wanted to try and be terser than lerna, which is why I went for project names without the `@typescript-eslint/` prefix for Nx, but that way is dumb because we miss out on the automatic dep-graph building from the package.json relationships (on top of the static imports).

When I address that, Nx's view of the dep-graph becomes:

![image](https://user-images.githubusercontent.com/900523/120217753-da8ce500-c249-11eb-96cc-141f0ca8eba3.png)

So it's just ast-utils that needs to be manually annotated because of that build and copy relationship.

Just let me know which projects implicitly depend on as-utils and I'll update it and push up the commit that generated the above

Reply:
It's just the types package which implicitly depends on it.

Everything else depends on the types package